### PR TITLE
LPS-201593 Set the className and classPK on the added file entry so that it will inherit the permissions

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related issue: [LPS-201593](https://liferay.atlassian.net/browse/LPS-201593)

### CODE REVIEW

**Comment by gabrielwas** [comment]
Could you implement integration tests for this behavior, please? Thx

**Review by G-Sa — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -1629,7 +1631,22 @@ private void _addFileEntry(
 				StringPool.BLANK, null, null, dlFileEntry.getContentStream(),
 				dlFileEntry.getSize(), null, null, serviceContext);
 
-			entry.setValue(fileEntry.getFileEntryId());
+			long fileEntryId = fileEntry.getFileEntryId();
+
+			DLFileEntry newDLFileEntry =
```
[the remark this answers is not in the document]
Comment (devpedrotavares) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java]: We agreed to go with another nomenclature

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
I added the integration tests [here](https://github.com/liferay-objects/liferay-portal/pull/2975/commits/7b4665940b8e5d4971a4af20df30f9fdf373828d)

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 2]
```diff
@@ -1589,8 +1590,9 @@ protected void modified(Map<String, Object> properties) {
 	}
 
 	private void _addFileEntry(
-			DLFileEntry dlFileEntry, Map.Entry<String, Serializable> entry,
-			long objectFieldId, List<ObjectFieldSetting> objectFieldSettings,
+			DLFileEntry tempDLFileEntry, Map.Entry<String, Serializable> entry,
+			long objectDefinitionId, long objectEntryId, long objectFieldId,
```
Comment (G-Sa) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1594]: Can you sort the variables by name?

**Review by G-Sa — CHANGES_REQUESTED** [review]
Hey @devpedrotavares, i noticed this after reviewing.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java (hunk 2 above)
Comment (devpedrotavares) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java]: I missed that after renaming. Thanks!

**Comment by devpedrotavares** [comment]
ci:test:sf

**Review by G-Sa — APPROVED** [review]
LGTM :tada:

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 3]
```diff
@@ -664,6 +665,101 @@ public void testAddObjectEntryAsDraft() throws Exception {
...
+		AssertUtils.assertFailure(
+			NoSuchFileEntryException.class,
+			StringBundler.concat(
+				"No FileEntry exists with the key {fileEntryId=",
+				tempFileEntryId, "}"),
+			() -> _dlAppLocalService.getFileEntry(tempFileEntryId));
+
+		Map<String, Serializable> values = objectEntry.getValues();
+
+		Assert.assertNotEquals(tempFileEntryId, values.get("upload"));
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:692]: @devpedrotavares use `MapUtil.getLong(objectEntry.getValues(), "upload");`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 4]
```diff
@@ -664,6 +665,101 @@ public void testAddObjectEntryAsDraft() throws Exception {
...
+
+		DLFileEntry persistedFileEntry = _dlFileEntryLocalService.getFileEntry(
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:694]: @devpedrotavares should be `persistedDLFileEntry`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 5]
```diff
@@ -664,6 +665,101 @@ public void testAddObjectEntryAsDraft() throws Exception {
...
+			(Long)values.get("upload"));
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:695]: @devpedrotavares use `MapUtil.getLong(objectEntry.getValues(), "upload");`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 6]
```diff
@@ -664,6 +665,101 @@ public void testAddObjectEntryAsDraft() throws Exception {
...
+						StringUtil.randomString());
+
+					return tempFileEntry2.getFileEntryId();
+				}
+			).build(),
+			ServiceContextTestUtil.getServiceContext());
+
+		values = objectEntry.getValues();
+
+		long persistedFileEntryId2 = (Long)values.get("upload");
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:748]: @devpedrotavares use `MapUtil.getLong(objectEntry.getValues(), "upload");`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 7]
```diff
@@ -664,6 +665,101 @@ public void testAddObjectEntryAsDraft() throws Exception {
...
+			ServiceContextTestUtil.getServiceContext());
+
+		AssertUtils.assertFailure(
+			NoSuchFileEntryException.class,
+			StringBundler.concat(
+				"No FileEntry exists with the key {fileEntryId=",
+				persistedFileEntryId1, "}"),
+			() -> _dlAppLocalService.getFileEntry(persistedFileEntryId1));
+
+		// Delete
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:731]: @devpedrotavares change to `// Delete object entry should delete existing files`

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 8]
```diff
@@ -1615,27 +1618,43 @@ private void _addFileEntry(
...
-			entry.setValue(fileEntry.getFileEntryId());
+			ObjectDefinition objectDefinition =
+				_objectDefinitionPersistence.fetchByPrimaryKey(
+					objectDefinitionId);
+
+			persistedDLFileEntry.setClassName(objectDefinition.getClassName());
+
+			persistedDLFileEntry.setClassPK(objectEntryId);
+
+			_dlFileEntryLocalService.updateDLFileEntry(persistedDLFileEntry);
```
[the remark this answers is not in the document]
Comment (devpedrotavares) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java]: This is much better, thank you!

**Comment by devpedrotavares** [comment]
ci:test:sf

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 9]
```diff
@@ -1602,10 +1605,10 @@ private void _addFileEntry(
 				return;
 			}
 
-			DLFolder dlFileEntryFolder = dlFileEntry.getFolder();
+			DLFolder dlFileEntryFolder = tempDLFileEntry.getFolder();
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1608]: @devpedrotavares please rename it back to `dlFileEntry`

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 10]
```diff
@@ -259,12 +259,12 @@ public ObjectEntry addObjectEntry(
...
 
-		_validateValues(
-			user.isGuestUser(), objectDefinitionId, null, serviceContext,
-			userId, values);
-
 		long objectEntryId = counterLocalService.increment();
 
+		_validateValues(
+			null, user.isGuestUser(), objectDefinitionId, objectEntryId,
+			serviceContext, userId, values);
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:266]: @devpedrotavares we already have the `objectDefinition` here

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 11]
```diff
@@ -366,8 +366,8 @@ public void addOrUpdateExtensionDynamicObjectDefinitionTableValues(
 		User user = _userLocalService.getUser(userId);
 
 		_validateValues(
-			user.isGuestUser(), objectDefinition.getObjectDefinitionId(), null,
-			serviceContext, userId, values);
+			null, user.isGuestUser(), objectDefinition.getObjectDefinitionId(),
+			primaryKey, serviceContext, userId, values);
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:370]: @devpedrotavares and here

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 12]
```diff
@@ -1454,8 +1454,9 @@ public ObjectEntry updateObjectEntry(
 				objectEntry.getObjectDefinitionId());
 
 		_validateValues(
-			user.isGuestUser(), objectEntry.getObjectDefinitionId(),
-			objectEntry, serviceContext, userId, values);
+			objectEntry, user.isGuestUser(),
+			objectEntry.getObjectDefinitionId(), objectEntryId, serviceContext,
+			userId, values);
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1459]: @devpedrotavares and here

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 13]
```diff
@@ -4504,6 +4469,8 @@ private void _validateValues(
 			return;
 		}
 
+		ObjectDefinition objectDefinition = objectField.getObjectDefinition();
+
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:4473]: @devpedrotavares so we don't need this, you can pass the `objectDefinition` to the `_validateValues` method

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**